browser/ui/webui/brave_new_tab/update_observer.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/brave_new_tab/update_observer.h"
+
+#include <utility>
+
+#include "brave/browser/ntp_background/ntp_background_prefs.h"
+#include "brave/components/ntp_background_images/common/pref_names.h"
+
+namespace brave_new_tab {
+
+UpdateObserver::UpdateObserver(PrefService& pref_service) {
```

Comment:
same as `CustomImageChooser`, why are we rewriting all of this? Per my previous comment I don't understand why we aren't just using a feature flag inside the existing ntp code. Based on the issue link it seems like it's maybe just a different design? Or I'm not even sure because there's no information in the issue other than it's supposed to allow choosing a background image which the current NTP already does